src/jit/lsrabuild.cpp

```diff
@@ -693,140 +970,34 @@ regMaskTP LinearScan::getKillSetForNode(GenTree* tree)
     regMaskTP killMask = RBM_NONE;
     switch (tree->OperGet())
     {
-#ifdef _TARGET_XARCH_
         case GT_MUL:
-            // We use the 128-bit multiply when performing an overflow checking unsigned multiply
-            //
-            if (((tree->gtFlags & GTF_UNSIGNED) != 0) && tree->gtOverflowEx())
-            {
-                // Both RAX and RDX are killed by the operation
-                killMask = RBM_RAX | RBM_RDX;
-            }
-            break;
-
         case GT_MULHI:
-#if defined(_TARGET_X86_) && !defined(LEGACY_BACKEND)
+#if !defined(_TARGET_64BIT_)
         case GT_MUL_LONG:
 #endif
-            killMask = RBM_RAX | RBM_RDX;
+            killMask = getKillSetForMul(tree->AsOp());
```

Comment:
I don't think it's necessary to maintain the kill mask as a class member. I'm making the general `getKillSetForNode` `DEBUG`-only; it will only be used in the assert and in the stress modes where we constrain nodes.